src/object_file/section.rs

```diff
@@ -0,0 +1,179 @@
+use std::marker::PhantomData;
+use std::ffi::CStr;
+use std::iter::FusedIterator;
+
+use llvm_sys::object::LLVMSectionIteratorRef;
+
+use super::{Binary, BinaryRef, RelocationIterator};
+
+#[derive(Debug)]
+pub struct Section<'a> {
+    section: LLVMSectionIteratorRef,
+    _marker: PhantomData<&'a Binary<'a>>,
+}
+impl<'a> Section<'a> {
+    pub(super) fn new(section: LLVMSectionIteratorRef) -> Self {
+        assert!(!section.is_null());
+
+        Self {
+            section,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn name(&self) -> &CStr {
+        use llvm_sys::object::LLVMGetSectionName;
+
+        unsafe {
+            CStr::from_ptr(LLVMGetSectionName(self.section))
+        }
+    }
+
+    pub fn address(&self) -> u64 {
+        use llvm_sys::object::LLVMGetSectionAddress;
+
+        unsafe {
+            LLVMGetSectionAddress(self.section)
+        }
+    }
+
+    pub fn size(&self) -> u64 {
+        use llvm_sys::object::LLVMGetSectionSize;
+
+        unsafe {
+            LLVMGetSectionSize(self.section)
+        }
+    }
+
+    pub fn contents(&self) -> &CStr {
+        use llvm_sys::object::LLVMGetSectionContents;
+
+        unsafe {
+            CStr::from_ptr(LLVMGetSectionContents(self.section))
+        }
+    }
+
+    pub fn relocations(&self) -> RelocationIterator<'a> {
+        use llvm_sys::object::LLVMGetRelocations;
+
+        let iter = unsafe {
+            LLVMGetRelocations(self.section)
+        };
+
+        RelocationIterator::new(iter, self.section)
+    }
+}
+
+
+#[derive(Debug)]
+pub struct SectionIterator<'a> {
+    binary: BinaryRef,
+    iter: LLVMSectionIteratorRef,
+    done: bool,
+    _marker: PhantomData<&'a Binary<'a>>,
+}
+impl<'a> SectionIterator<'a> {
+    #[llvm_versions(9.0..=latest)]
+    pub(super) fn new(binary: BinaryRef) -> Self {
```

Comment:
Since both versions of `new` have the same signature we should consolidate them into one function with `cfg`s on the call site(s).

Same with the other methods below